Dockerfile

```diff
@@ -0,0 +1,27 @@
+ARG ROS_DISTRO=rolling
+FROM ros:${ROS_DISTRO}-ros-core
+
+RUN apt-get update \
+    && apt-get install -y \
+        ros-dev-tools \
+        wget
+        
+WORKDIR /root/ros2_ws/
+RUN mkdir -p src
+COPY tools/ros2_dependencies.repos .
+RUN vcs import --input ros2_dependencies.repos src
+RUN rosdep init
+
+COPY . src/grid_map
+RUN ls src/grid_map
+
+SHELL ["/bin/bash", "-c"]
+RUN apt-get update \
+    && rosdep update
+
+RUN source /opt/ros/${ROS_DISTRO}/setup.bash \
+    && rosdep install -y --ignore-src --from-paths src --skip-keys slam_toolbox
```

Comment:
@stevemacenski This is a proposed workaround in `grid_map` to deal with the fact that not all the keys resolve in `nav2` when we build it from source in `rolling`.
Do you have any way in NAV2 to ensure that there won't be new keys added in `rolling` that don't resolve? If that happens, it would break CI here.

If I use the `-r` flag globally, then I let contributors to grid_map contribute mis-typed keys. This is a real problem and happened in ArduPilot before: 
https://github.com/ArduPilot/ardupilot_ros/pull/23

Reply:
Are you only building nav2 for costmap2d? If so, you could give rosdep only that package path to resolve and build only up to that.

Reply:
Both `costmap2d` and `nav2_msgs` are used in `grid_map`. Sounds like the risk of breakage to this line of code is very low.